src/avx512bw/avx512bw-from-utf8.inl.cpp

```diff
@@ -0,0 +1,466 @@
+// file included directly
+
+// File contains conversion procedure from possibly invalid UTF-8 strings.
+
+/*
+validate_utf8_structure only checks if leading bytes
+are followed by proper number of continuation bytes
+and then there's another leading byte.
+
+We don't handle end-of-stream here.
+*/
+bool validate_utf8_structure(__m512i input) {
```

Comment:
I think that this may not catch a sequence that begins with a continuation byte, would it?

Reply:
Right, we assume that the stream starts with a leading byte. There's my omission here, as I have this check in the proof-of-code. Will add it.

Reply:
Added the missing check